core/domain/translation_domain.py

```diff
@@ -18,9 +18,287 @@
 
 from __future__ import annotations
 
+import copy
+import enum
+
 from core import utils
 
-from typing import Dict
+from typing import Dict, List, Union
+from typing_extensions import TypedDict
+
+
+class TranslatableContentPossibleDataTypes(enum.Enum):
+    """Represents all possible data types for any translatable content."""
+
+    TRANSLATABLE_CONTENT_FORMAT_HTML = 'html'
+    TRANSLATABLE_CONTENT_FORMAT_UNICODE_STRING = 'unicode'
+    TRANSLATABLE_CONTENT_FORMAT_SET_OF_NORMALIZED_STRING = (
+        'set_of_normalized_string')
+    TRANSLATABLE_CONTENT_FORMAT_SET_OF_UNICODE_STRING = 'set_of_unicode_string'
+
+
+class EntityTypesSupportingNewTranslationArch(enum.Enum):
```

Comment:
I'm not sure this is the right name -- we should always have a single translation architecture. That's why we're migrating explorations to it before starting with the other entities. So maybe the name of this should be something like TranslatableEntityType instead.

Reply:
Updated